### PR TITLE
Kube helper function refactor and unit tests

### PR DESCRIPTION
**Pull request by josephsawaya**
This PR refactors the kube method helper functions and adds unit tests for those functions.

Depends on #123 

Signed-off-by: Joseph Sawaya <jsawaya@redhat.com>

### CODE REVIEW

**Comment by cooktheryan** [comment]
@sallyom this is covered in your PR as well

**Comment by sallyom** [comment]
This PR was pulled in with https://github.com/redhat-et/harpoon/pull/127
Thanks, @josephsawaya!

File: pkg/engine/kube.go [hunk 1]
```diff
@@ -19,149 +21,183 @@ import (
...
+func getPodNames(podList []v1.Pod) ([]string, error) {
 	ret := make([]string, 0)
-	if err != nil {
-		return nil, err
-	}
-	for _, pod := range pod_list {
-		ret = append(ret, pod.ObjectMeta.Name)
+
+	for _, pod := range podList {
+		if pod.ObjectMeta.Name != "" {
```
Comment (jeffvance) [inline pkg/engine/kube.go:153]: This is a real nit: imo it's slightly more idiomatic to refactor as:
```
for ... {
     if pod.ObjectMeta.Name == "" {
           return nil, errors.New(...)
     }
     ret = append(...) 
}
```

File: pkg/engine/kube_test.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	pod.Spec.Containers[0].Name = pod.ObjectMeta.Name
+
+	err = validatePod(pod)
+	if err == nil {
+		t.Errorf("expected error, got nil")
+	}
+
+}
+
+var invalidYaml = `
```
Comment (jeffvance) [inline pkg/engine/kube_test.go:49]: perhaps a bit more idiomatic is to not use `var`, instead:
```
 invalidYaml :=   ....

File: pkg/engine/kube_test.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+metadata:
+  name: test_pod2
+spec:
+  containers:
+  - name: test2
+    image: test2`
+
+var expectedPodList = []v1.Pod{
+	{
+		TypeMeta: metav1.TypeMeta{
```
Comment (jeffvance) [inline pkg/engine/kube_test.go:79]: It is more common in k8s to omit the two embedded types - TypeMeta and ObjectMeta, eg:
```
Kind: "Pod",
APIVersion "v1",
Name: "test_pod1",
...
```

File: pkg/engine/kube_test.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+	if len(expected) != len(actual) {
+		t.Errorf("expected %d pods, got %d", len(expected), len(actual))
+	}
+
+	for _, pod := range expected {
+		pod_bytes, err := yaml.Marshal(pod)
+		if err != nil {
+			t.Errorf("error running test: %s", err)
+		}
+		if _, ok := m[string(pod_bytes)]; ok {
```
Comment (jeffvance) [inline pkg/engine/kube_test.go:125]: where would this be true? Won't the uuid always be different between 2 pods?

**Pull request closed without merging**